### PR TITLE
Improved how current cluster member role is determined

### PR DESCRIPTION
**Pull request by lutovich**
This PR removes some code needed for rolling upgrades since those are no longer supported and improves how cluster member roles are determined.

Signed-off-by: @MishaDemianenko

### CODE REVIEW

**Comment by lutovich** [comment]
@digitalstain, @tinwelint could you please review?

**Comment by tinwelint** [comment]
I'm enjoying the descriptive commit message and javadocs, sets the stage nicely. Changes look good.

**Comment by tinwelint** [comment]
How is verifications of this fix going?

**Comment by lutovich** [comment]
Running tests for HA robustness now.

File: enterprise/ha/src/main/java/org/neo4j/kernel/ha/cluster/member/ClusterMembers.java [hunk 1]
```diff
@@ -19,40 +19,22 @@
...
+import org.neo4j.kernel.ha.cluster.HighAvailabilityMemberStateMachine;
 import org.neo4j.kernel.ha.cluster.HighAvailabilityModeSwitcher;
-import org.neo4j.kernel.impl.store.StoreId;
-import org.neo4j.kernel.impl.util.CopyOnWriteHashMap;
 
 /**
- * Keeps an up to date list of members, their roles and availability for
- * display for example in JMX.
+ * Keeps a list of members, their roles and availability for display for example in JMX or REST.
+ * <p/>
```
Comment (davidegrohmann) [inline enterprise/ha/src/main/java/org/neo4j/kernel/ha/cluster/member/ClusterMembers.java]: Please don't add `<p/>`, they will break java 8 javadocs.

File: enterprise/ha/src/main/java/org/neo4j/kernel/ha/cluster/member/ObservedClusterMembers.java [hunk 2]
```diff
@@ -0,0 +1,218 @@
...
+import org.neo4j.kernel.impl.util.CopyOnWriteHashMap;
+import org.neo4j.kernel.impl.util.StringLogger;
+import org.neo4j.kernel.logging.Logging;
+
+/**
+ * Keeps list of members, their roles and availability.
+ * List is based on different notifications about cluster and HA events.
+ * List is basically a 'best guess' of the cluster state because message ordering is not guaranteed.
+ * This class should be used only when imprecise state is acceptable.
+ * <p/>
```
Comment (davidegrohmann) [inline enterprise/ha/src/main/java/org/neo4j/kernel/ha/cluster/member/ObservedClusterMembers.java]: java 8 javadocs

**Pull request merged**